example/module_sensor_data_publisher.c

```diff
@@ -7,11 +7,13 @@
 #include "FreeRTOS.h"
 #include "timers.h"
 
-#include "eventrouter/event.h"
-#include "eventrouter/event_handler.h"
-#include "eventrouter/eventrouter.h"
```

Comment:
Why are you removing the includes? At least some of them are used in the code. I see that they are included through the `module_sensor_data_publisher.h` (some of them, like `module.h` is further indirectly through other includes), but should that be the intent?

If you want the users to use just one include, like `eventrouter.h`, then maybe "hide" other includes into the "internal" folder? Otherwise maybe consider Include-What-You-Use.

Reply:
These are good observations. I think I'm going to go with the single "eventrouter.h" for clients and make all the component-headers internal only.